### PR TITLE
Append the current session identifier to the protocol name prefix

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This should simplify debugging of parallel drivers

### CODE REVIEW

File: lib/protocol/protocol.js [hunk 1]
```diff
@@ -35,31 +36,61 @@ const IMG_EL_URL_RE = new RegExp(
...
+  getLogger (sessionId, protocol) {
+    if (sessionId) {
+      if (this._cache.has(sessionId)) {
+        const value = this._cache.get(sessionId);
+        if (value.logger) {
+          return value.logger;
+        }
+        protocol = protocol || value.protocol;
+      }
+      return logger.getLogger(`${protocol || GENERIC_PROTOCOL} ` +
```
Comment (imurchie) [inline lib/protocol/protocol.js]: Should this get added to `value`? If, for whatever reason, there is no logger for a session id, each log line will create a new logger, which is then thrown out.

Comment (mykola-mokhnach) [reply lib/protocol/protocol.js]: This was done intentionally to include the "wrong" session identifier into the actual log lines without caching the whole logger. 

Although, if we don't want to create a new instance each time this method is called, then we could either fall back to the default logger with protocol name only or include some generic prefix there, like `[W3C (unknown session)]`.

Comment (mykola-mokhnach) [reply lib/protocol/protocol.js]: We don't want to put each random session id into the cache in order to save memory. Instead we only put these ids that are returned by `createSession` call

Comment (imurchie) [reply lib/protocol/protocol.js]: Ok. This should be documented, or else it will be "fixed" at some later date.

Comment (mykola-mokhnach) [reply lib/protocol/protocol.js]: I'll add a comment

**Pull request merged**